### PR TITLE
add flareWith

### PR DESCRIPTION
**Pull request by rintcius**
Hi @sharkdp, first of all thanks for this great library!

I have added a `flareWith` function which enables hooking `Flare`s without running the signal. It is just a slightly more generic version of `runFlareWith` (and is now written in terms of `flareWith`).

I'd be happy to know your thoughts about this PR.

### CODE REVIEW

**Comment by sharkdp** [comment]
Thank you very much for the PR!

I'll review this in more detail next week, when I am back home, but looks good to me so far!

Could give an example for when this might be useful?

**Comment by rintcius** [comment]
Ok, great!

I need it to integrate with non Flare components that expect `Signal` parameters. E.g given:

```
myNumberFlare :: forall e.UI e Number
myComponent :: Signal Number -> Eff _ Unit
```

I can then use it like this (`myComponent` reacting to changes in `myNumberFlare`):

```
main = do
  flareWith "controls" myComponent myNumberFlare
```

File: src/Flare.purs [hunk 1]
```diff
@@ -387,16 +388,25 @@ setupFlare (UI setupUI) = do
...
 runFlareWith :: forall e a. ElementId
              -> (a -> Eff (dom :: DOM, channel :: CHANNEL | e) Unit)
              -> UI e a
              -> Eff (dom :: DOM, channel :: CHANNEL | e) Unit
-runFlareWith controls handler (UI setupUI) = do
-  (Flare components signal) <- setupUI
-  removeChildren controls
-  traverse_ (appendComponent controls) components
-  S.runSignal (map handler signal)
+runFlareWith controls handler ui = flareWith controls (\sig -> S.runSignal (map handler sig)) ui
```
Comment (sharkdp) [inline src/Flare.purs]: I believe the lambda could be replaced by `S.runSignal <<< map handler`?

Comment (rintcius) [reply src/Flare.purs]: Indeed, looks like it can. I'll have a look at it tonight and test it.

**Comment by rintcius** [comment]
@sharkdp Updated and tested

**Comment by sharkdp** [comment]
Thanks! I've released a new version.

**Comment by rintcius** [comment]
Great, thanks you!

**Pull request merged**